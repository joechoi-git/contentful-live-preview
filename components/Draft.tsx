import React from "react";
import { draftMode } from "next/headers";

export default function Draft() {
    const { isEnabled } = draftMode();
    return (
        <div
            id="control-panel"
            className="my-6 border-4 border-accent bg-primary text-primary-content rounded w-full p-4 flex gap-4"
        >
            <p className="font-bold">Control Panel</p>
            {isEnabled ? (
                <p className="text-primary-content">
                    Draft Mode is <strong>Enabled.</strong>{" "}
                    <a href="/api/disable-draft-mode" className="text-primary-content underline">
                        Disable Draft Mode
                    </a>
                </p>
            ) : (
                <p className="text-secondary-content">
                    Draft Mode is <strong>Disabled.</strong>{" "}
                    <a href="/api/enable-draft-mode" className="text-primary-content underline">
                        Enable Draft Mode
                    </a>
                </p>
            )}
        </div>
    );
}
