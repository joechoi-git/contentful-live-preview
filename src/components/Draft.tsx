import React from "react";
import { draftMode } from "next/headers";

export default function Draft() {
    const { isEnabled } = draftMode();
    return (
        <div
            id="control-panel"
            className="my-6 border border-blue-500 rounded w-full p-4 flex gap-4 bg-slate-100"
        >
            <p className="font-bold">Control Panel</p>
            {isEnabled ? (
                <p className="text-green-500">
                    Draft Mode is <strong>Enabled.</strong>{" "}
                    <a href="/api/disable-draft-mode" className="text-blue-500 underline">
                        Disable Draft Mode
                    </a>
                </p>
            ) : (
                <p className="text-red-500">
                    Draft Mode is <strong>Disabled.</strong>{" "}
                    <a href="/api/enable-draft-mode" className="text-blue-500 underline">
                        Enable Draft Mode
                    </a>
                </p>
            )}
        </div>
    );
}
