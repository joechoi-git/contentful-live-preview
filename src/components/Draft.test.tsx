import React from "react";
import { render, screen } from "@testing-library/react";
import Draft from "../components/Draft";

jest.mock("next/headers", () => ({
    draftMode: jest.fn()
}));

describe("Draft Component", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test("renders correctly when draft mode is enabled", () => {
        const draftModeMock = require("next/headers").draftMode;
        draftModeMock.mockReturnValue({ isEnabled: true });

        const { asFragment } = render(<Draft />);
        expect(screen.getByText("Control Panel")).toBeInTheDocument();
        expect(screen.getByText("Draft Mode is")).toBeInTheDocument();
        expect(screen.getByText("Enabled.")).toBeInTheDocument();
        expect(screen.getByText("Disable Draft Mode")).toHaveAttribute(
            "href",
            "/api/disable-draft-mode"
        );

        expect(asFragment()).toMatchSnapshot();
    });

    test("renders correctly when draft mode is disabled", () => {
        const draftModeMock = require("next/headers").draftMode;
        draftModeMock.mockReturnValue({ isEnabled: false });

        const { asFragment } = render(<Draft />);
        expect(screen.getByText("Control Panel")).toBeInTheDocument();
        expect(screen.getByText("Draft Mode is")).toBeInTheDocument();
        expect(screen.getByText("Disabled.")).toBeInTheDocument();
        expect(screen.getByText("Enable Draft Mode")).toHaveAttribute(
            "href",
            "/api/enable-draft-mode"
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
