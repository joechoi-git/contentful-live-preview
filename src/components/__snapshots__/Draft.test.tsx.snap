// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draft Component renders correctly when draft mode is disabled 1`] = `
<DocumentFragment>
  <div
    class="my-6 border-4 border-accent bg-primary text-primary-content rounded w-full p-4 flex gap-4"
    id="control-panel"
  >
    <p
      class="font-bold"
    >
      Control Panel
    </p>
    <p
      class="text-secondary-content"
    >
      Draft Mode is 
      <strong>
        Disabled.
      </strong>
       
      <a
        class="text-primary-content underline"
        href="/api/enable-draft-mode"
      >
        Enable Draft Mode
      </a>
    </p>
  </div>
</DocumentFragment>
`;

exports[`Draft Component renders correctly when draft mode is enabled 1`] = `
<DocumentFragment>
  <div
    class="my-6 border-4 border-accent bg-primary text-primary-content rounded w-full p-4 flex gap-4"
    id="control-panel"
  >
    <p
      class="font-bold"
    >
      Control Panel
    </p>
    <p
      class="text-primary-content"
    >
      Draft Mode is 
      <strong>
        Enabled.
      </strong>
       
      <a
        class="text-primary-content underline"
        href="/api/disable-draft-mode"
      >
        Disable Draft Mode
      </a>
    </p>
  </div>
</DocumentFragment>
`;
